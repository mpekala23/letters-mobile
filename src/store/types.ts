import { UserState } from "./User/UserTypes";
import { NotifState } from "./Notif/NotifTypes";
import { ContactState } from "./Contact/ContactTypes";

export interface AppState {
  user: UserState;
  notif: NotifState;
  contact: ContactState;
}
