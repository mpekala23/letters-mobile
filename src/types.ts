// TODO: make this typing better
// probably want a specific type for state abbrevs
export interface Facility {
  name: string;
  type: PrisonTypes;
  address: string;
  city: string;
  state: string;
  postal: string;
  phone: string;
}

export enum ProfilePicTypes {
  Topbar = 'Topbar',
  Contact = 'Contact',
  SingleContact = 'SingleContact',
}

export enum Storage {
  RememberToken = 'Ameelio-Token',
}

export enum PrisonTypes {
  State = 'State',
  Federal = 'Federal',
  Jail = 'County',
}

export enum SupportFAQTypes {
  DeleteLetter = 'DeleteLetter',
  NotArrived = 'NotArrived',
  WrongReturnAddress = 'WrongReturnAddress',
  WrongMailingAddress = 'WrongMailingAddress',
  TrackingNumber = 'TrackingNumber',
  TrackingError = 'TrackingError',
  TalkToAmeelio = 'TalkToAmeelio',
}

export enum LetterStatus {
  Draft = 'Draft',
  Created = 'Created',
  Mailed = 'Mailed',
  InTransit = 'In Transit',
  InLocalArea = 'In Local Area',
  OutForDelivery = 'Out for Delivery',
  Delivered = 'Delivered',
}

export enum LetterTypes {
  Postcard = 'postcard',
  Letter = 'letter',
}

export enum DeliveryReportTypes {
  received = 'received',
  unsure = 'unsure',
  notYetReceived = 'notYetReceived',
  haveNotAsked = 'haveNotAsked',
  haveNotReceived = 'haveNotReceived',
}

export interface Letter {
  type: LetterTypes;
  status: LetterStatus;
  isDraft: boolean;
  recipientId: number;
  content: string;
  photo?: Photo;
  letterId?: number; // TODO: Once we have more info on this field and lob, use this more
  expectedDeliveryDate?: Date;
  dateCreated?: Date;
  trackingEvents?: LetterTrackingEvent[];
}

export interface LetterTrackingEvent {
  id: number;
  name: string;
  location: ZipcodeInfo;
  date: Date;
}

export type TopbarRouteAction = {
  enabled: boolean;
  text: string;
  action: () => void | Promise<void>;
  blocking?: boolean;
};

export interface ZipcodeInfo {
  zip: string;
  city: string;
  state: string;
  lat?: number;
  long?: number;
}

export interface Photo {
  type?: 'image' | string;
  width?: number;
  height?: number;
  uri: string;
}
