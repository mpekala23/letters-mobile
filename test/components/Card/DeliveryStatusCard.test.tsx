import React from "react";
import { DeliveryStatusCard } from "@components";
import { fireEvent, render, toJSON } from "@testing-library/react-native";
import { LetterStatus } from "types";

const setup = (propOverrides = {}) => {
  const props = Object.assign(
    {
      title: "Title",
      status: "Status",
      date: "Date",
      progress: 0,
      onPress: jest.fn(),
    },
    propOverrides
  );
  return {
    ...render(<DeliveryStatusCard {...props} />),
    props,
  };
};

describe("Delivery Status Card component", () => {
  it("should match snapshot", () => {
    const { container } = setup();
    const tree = toJSON(container);
    expect(tree).toMatchSnapshot();
  });

  it("should fire onPress() on a press", () => {
    const { props, getByText } = setup();
    fireEvent.press(getByText("Title"));
    expect(props.onPress).toHaveBeenCalledTimes(1);
  });

  it("should display title", () => {
    const { getByText } = setup();
    expect(getByText("Title")).toBeDefined();
  });

  it("should display status", () => {
    const { getByText } = setup();
    expect(getByText("Status")).toBeDefined();
  });

  it("should display date", () => {
    const { getByText } = setup();
    expect(getByText("Date")).toBeDefined();
  });

  it("should display progress bar 0 / 5", () => {
    const { getByTestId } = setup({ status: LetterStatus.Draft });
    expect(getByTestId("progressBar").props.style[1].width).toBe("0%");
  });

  it("should display progress bar 1 / 5", () => {
    const { getByTestId } = setup({ status: LetterStatus.Created });
    expect(getByTestId("progressBar").props.style[1].width).toBe("20%");
  });

  it("should display progress bar 2 / 5", () => {
    const { getByTestId } = setup({ status: LetterStatus.Printed });
    expect(getByTestId("progressBar").props.style[1].width).toBe("40%");
  });

  it("should display progress bar 3 / 5", () => {
    const { getByTestId } = setup({ status: LetterStatus.Mailed });
    expect(getByTestId("progressBar").props.style[1].width).toBe("60%");
  });

  it("should display progress bar 4 / 5", () => {
    const { getByTestId } = setup({
      status: LetterStatus.OutForDelivery,
    });
    expect(getByTestId("progressBar").props.style[1].width).toBe("80%");
  });

  it("should display progress bar 5 / 5", () => {
    const { getByTestId } = setup({ status: LetterStatus.Delivered });
    expect(getByTestId("progressBar").props.style[1].width).toBe("100%");
  });
});
