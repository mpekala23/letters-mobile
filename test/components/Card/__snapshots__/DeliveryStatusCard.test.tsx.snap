// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Delivery Status Card component should match snapshot 1`] = `
<View
  collapsable={true}
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <TouchableOpacity
    activeOpacity={0.2}
    style={
      Array [
        Object {
          "backgroundColor": "white",
          "borderRadius": 6,
          "marginVertical": 6,
          "padding": 15,
          "width": "100%",
        },
        Object {
          "elevation": 4,
          "shadowColor": "#000",
          "shadowOffset": Object {
            "height": 2,
            "width": 0,
          },
          "shadowOpacity": 0.23,
          "shadowRadius": 2.62,
        },
        undefined,
      ]
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "space-between",
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#282828",
            "fontSize": 24,
            "fontWeight": "bold",
          }
        }
      >
        Title
      </Text>
      <Text
        style={
          Object {
            "color": "#6E757C",
            "fontSize": 16,
          }
        }
      >
        Date
      </Text>
    </View>
    <Text
      style={
        Object {
          "color": "#6E757C",
          "fontSize": 18,
        }
      }
    >
      Status
    </Text>
    <View
      style={
        Object {
          "backgroundColor": "#F2F2F2",
          "borderRadius": 5,
          "height": 10,
          "marginTop": 15,
          "overflow": "hidden",
          "width": "100%",
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "backgroundColor": "#6D89B1",
              "height": 10,
              "position": "absolute",
            },
            Object {
              "width": 0,
            },
          ]
        }
        testID="progressBar"
      />
    </View>
  </TouchableOpacity>
</View>
`;
